maas/resource_maas_static_route.go

```diff
@@ -0,0 +1,187 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMaasStaticRoute() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS static routes.",
+		CreateContext: resourceStaticRouteCreate,
+		ReadContext:   resourceStaticRouteRead,
+		UpdateContext: resourceStaticRouteUpdate,
+		DeleteContext: resourceStaticRouteDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				cfg := meta.(*ClientConfig)
+				client := cfg.Client
+				static_route, err := getStaticRoute(client, d.Id())
+				if err != nil {
+					return nil, err
+				}
+				tfState := map[string]interface{}{
+					"id":          fmt.Sprintf("%v", static_route.ID),
+					"source":      static_route.Source.Name,
+					"destination": static_route.Destination.Name,
+					"gateway_ip":  static_route.GatewayIP,
+					"metric":      static_route.Metric,
+				}
+				if err := setTerraformState(d, tfState); err != nil {
+					return nil, err
+				}
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
```

Comment:
Could you reorder these so the linter is happy?
`destination`, `gateway_ip`, `metric`, and then `source`?